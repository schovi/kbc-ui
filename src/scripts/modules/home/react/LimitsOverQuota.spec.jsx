import React from 'react';
import Immutable from 'immutable';
import LimitsOverQuota from './LimitsOverQuota';

describe('<LimitsOverQuota />', function() {
  it('should render nothing for no limits', function() {
    const limits = Immutable.List();
    shallowSnapshot(<LimitsOverQuota limits={limits} />);
  });

  it('should render limits', function() {
    const limits = Immutable.fromJS([
      {
        unit: 'bytes',
        metricValue: 12884901888, // 12GB
        limitValue: 107374182400,
        section: 'Section 1',
        name: 'My Limit 1'
      },
      {
        metricValue: 1100,
        limitValue: 15000,
        section: 'Section 2',
        name: 'My Limit 2'
      }
    ]);

    shallowSnapshot(<LimitsOverQuota limits={limits} />);
  });
});