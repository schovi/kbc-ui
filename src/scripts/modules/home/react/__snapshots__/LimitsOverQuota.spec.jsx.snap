exports[`<LimitsOverQuota /> should render limits 1`] = `
<div
  className="row kbc-header kbc-limits">
  <div
    className="alert alert-danger">
    <h3>
      <span
        className="fa fa-exclamation-triangle" />
       Project is over quota!
    </h3>
    <ul>
      <li>
        <strong>
          Section 1
           - 
          My Limit 1
        </strong>
         
        (12.88 GB of 107.37 GB)
      </li>
      <li>
        <strong>
          Section 2
           - 
          My Limit 2
        </strong>
         
        (1,100 of 15,000)
      </li>
    </ul>
    <p>
      <Link
        activeClassName="active"
        className=""
        to="settings-limits">
        Limits Settings
      </Link>
    </p>
  </div>
</div>
`;

exports[`<LimitsOverQuota /> should render nothing for no limits 1`] = `null`;
